src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs

```diff
@@ -4723,8 +4743,13 @@ public static void WaitAll(params Task[] tasks)
         [MethodImpl(MethodImplOptions.NoOptimization)]  // this is needed for the parallel debugger
         public static bool WaitAll(Task[] tasks, TimeSpan timeout)
         {
+            if (tasks is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.tasks);
+            }
+
```

Comment:
Duplicated check - I missed it when resolving conflicts. FWIW: this is the only file that had conflicts, most likely because https://github.com/dotnet/runtime/pull/100569 got merged in between.